src/foraging_gui/Foraging.py

```diff
@@ -94,6 +94,7 @@ def __init__(self, parent=None,box_number=1,start_bonsai_ide=True):
         self.ToInitializeVisual = 1 # Should we visualize performance
         self.FigureUpdateTooSlow = 0# if the FigureUpdateTooSlow is true, using different process to update figures
         self.ANewTrial = 1          # permission to start a new trial
+        self.to_save = 1            # permission to save backup data
```

Comment:
please use more explicit name such as `save_backup_allowed`, although I don't quite understand the function of this flag. Could you add more documentation?

Reply:
I updated the name to `save_backup_allowed`. 

The backup saving is running in a different thread. When  `save_backup_allowed == 0`, indicating that the previous saving has not finished, and it will not trigger the next saving; `save_backup_allowed == 1`, it is allowed to save backup data.